lib/search.rb

```diff
@@ -17,8 +26,25 @@ def create_search_index
       end
     end
 
+    JSON.generate(index)
+  end
+
+  def write_json_file(json)
     index_file = File.join(@config[:output_dir], "search.json")
-    File.write(index_file, JSON.generate(index))
+    File.write(index_file, json)
+  end
+
+  def write_js_file(json)
+    js_file = File.join(@config[:output_dir], "search/index.js")
+    js_lines = File.open(js_file).readlines.map(&:chomp)
+    js_lines[1] = "  var ARTICLES = #{json};"
+    js_lines[2] = "  var DICTIONARY = #{dictionary};"
+    File.write(js_file, js_lines.join("\n"))
+  end
```

Comment:
This seems very brittle. A change to the .js file may screw up this method. But that dependency is pretty much invisible. Could we achieve the same using ERB?

Something along the lines of 

```ruby
require 'erb'

simple_template = File.read(js_file)

renderer = ERB.new(simple_template)
File.write(js_file,renderer.result())
```

Reply:
Alternativly passing in `ARTICLES` and `DICTIONARY` via json as well.

Reply:
Great idea – I like that mucho better! Thanks!

Reply:
I did not realize there was a Netlify preview link in the branch checks. 🤯

Reply:
So, I had an awful time implementing this because I was developing it by visiting `search.js` in a browser. It took me _too_ long to figure out there is a weird browser encoding quirk that encodes special characters when accessing a `.js` file directly. Everything works fine, just the source code looks funny. Of course, the code works when included in an HTML file though. 🤷